openedx/openedx-platform
# [BB-6151] feat: upgrades tinyMCE v4.0.20 to tinyMCE v5.5.1

## Description

This PR upgrades TinyMCE version 4 to version 5.5.1.

These changes will impact the "Course author".

## Supporting information
- [Discovery doc to upgrade TinyMCE **[Approved]**](https://docs.google.com/document/d/1glPBW6OYkhkn_jAJndC8JMyruHl4RY9YGd4bqWbySg0/edit)
- Internal ticket: https://tasks.opencraft.com/browse/BB-6151

## Testing instructions

#### Testing new TinyMCE in Studio
1. Run the studio server from devstack dir “make studio-up”
1. Login to studio and create a new course.
1. Add a new unit in the course and go to the unit editor page.
1. Click “HTML” component button and select “Text”.
1. Click the component to edit, this will show a modal containing “tinymce” editor.
1. Use the TinyMCE editor thoroughly to test all the features.
1. Save the changes and expect it to get saved and loads correctly after refreshing the page.

#### Ensure the “visual” and “raw” html editor are working fine
1. Run the studio server from devstack dir “make studio-up”
1. Login to studio and create a new course.
1. Add a new unit in the course and go to the unit editor page.
1. Click “HTML” component button and select “Raw”.
1. Click the component to edit, this will show a modal containing “codemirror” editor.
1. Click the “setting-icon” in the editor modal and change the “Editing mode” to “visual”.
1. Click the component to edit, this will show a modal containing “tinyMce” editor.

#### Ensure the “codemirror-plugin” works fine in the tinymce editor
1. Run the studio server from devstack dir “make studio-up”
1. Login to studio and create a new course.
1. Add a new unit in the course and go to the unit editor page.
1. Click “HTML” component button and select “Text”.
1. Click the component to edit, this will show a modal containing “tinymce” editor.
1. Click on “HTML” icon on the toolbar of the editor, this will show a modal with codemirror editor.
1. Add some HTML content and save the changes.
1. The saved changes should show up in the TinyMCE editor.

#### Testing announcements message in studio
1. Run the studio server from devstack dir “make studio-up”
1. Go to “/maintenance” and select “announcement” section.
1. Click “Create new announcement” button.
1. The TinyMCE editor will show up, make sure to test all the components in the editor.
1. Save the announcement message.

## Screenshots and videos of UI changes
**Video comparing TinyMCE4 and TinyMCE5 in studio**

https://user-images.githubusercontent.com/16653571/166076672-72165416-fdc6-49ad-a64c-0a8139fc525b.mp4

#### OLD TinyMCE UI in Studio
![image](https://user-images.githubusercontent.com/16653571/166076360-dc2da1d9-2bcf-4ed7-80d9-406d7d8744b3.png)
#### NEW TinyMCE UI in Studio
![image](https://user-images.githubusercontent.com/16653571/166076334-aa3ce673-1afb-4d31-b7dc-40793d12a551.png)
#### OLD TinyMCE UI on announcement editor page
![image](https://user-images.githubusercontent.com/16653571/166077813-a6f29de0-6ca4-4b8a-975e-0b409c966781.png)
#### NEW TinyMCE UI on announcement editor page
![image](https://user-images.githubusercontent.com/16653571/166077969-96bfbba1-13cc-4c7c-a8ae-1dc97fb8bfab.png)
